src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        final Node<T> newNode = new Node<>(value, tail, null);
+        if (tail != null) {
+            tail.next = newNode;
+        }
+        checkHeadTail(newNode);
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> replacedNode;
+        Node<T> newNode;
+        if (index == size) {
+            newNode = new Node<>(value, tail, null);
+        } else {
+            replacedNode = getAt(index);
+            if (replacedNode != null) {
+                newNode = new Node<>(value, replacedNode.previous, replacedNode);
+                replacedNode.previous = newNode;
+            } else {
+                newNode = new Node<>(value, null, null);
+            }
+        }
+        if (newNode.previous != null) {
+            newNode.previous.next = newNode;
+        }
+        checkHeadTail(newNode);
+        size++;
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public void addAll(List<T> values) {
+        for (T toAdd : values) {
+            add(toAdd);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getAt(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        final Node<T> foundNode = getAt(index);
+        final T oldValue = foundNode.value;
+        foundNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> foundNode = getAt(index);
+        unlinc(foundNode);
+        return foundNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        for (int i = 0; i < size; i++) {
+            T value = current.value;
+            if (value == null && object == null
+                    || value != null && value.equals(object)) {
+                unlinc(current);
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
+    private void unlinc(Node<T> foundNode) {
```

Comment:
All private methods should e after all public methods. 
Be careful with typo mistakes in the name of the method.
```suggestion
    private void unlink(Node<T> foundNode) {
```